relay-sampling/src/evaluation.rs

```diff
@@ -41,9 +147,16 @@ impl SamplingEvaluator {
             rule_ids: vec![],
             factor: 1.0,
             client_sample_rate: None,
+            reservoir: None,
         }
     }
 
+    /// Sets a [`ReservoirEvaluator`].
+    pub fn set_reservoir(mut self, reservoir: Option<Arc<ReservoirEvaluator>>) -> Self {
```

Comment:
_nit_: What if we change the signature:

```suggestion
    pub fn set_reservoir(mut self, reservoir: Arc<ReservoirEvaluator>) -> Self {
```

Or in combination with my other comment:

```rust
impl <'a> SamplingEvaluator<'a> {
    // ...
    pub fn set_reservoir(mut self, reservoir: &'a mut ReservoirEvaluator) -> Self {
```